apps/tasks/models.py

```diff
@@ -1,45 +1,104 @@
-    assigned_to = models.ForeignKey(User, on_delete=models.CASCADE, related_name='owner', null=True)
-    created_by = models.ForeignKey(User, on_delete=models.CASCADE, related_name='created_by', null=True)
-
-    def __str__(self):
-        return f" \"{self.title}\" By: {self.assigned_to.get_full_name()}"
+    description = models.TextField(blank=True)
+    status = models.CharField(max_length=50, choices=Status.choices, default=Status.TODO)
+    assigned_to = models.ForeignKey(User, on_delete=models.CASCADE, related_name='assigns', null=True)
+    created_by = models.ForeignKey(User, on_delete=models.CASCADE, related_name='created_tasks', null=True)
 
 
 class Comment(models.Model):
     task = models.ForeignKey(Task, on_delete=models.CASCADE)
-    text = models.CharField(max_length=255)
+    text = models.TextField()
+
+    class Meta:
+        ordering = ['-id']
 
     def __str__(self):
         return f'{self.task.title} : {self.text}'
 
 
-class Timelog(models.Model):
-    task = models.ForeignKey(Task, on_delete=models.CASCADE)
-    start = models.DateTimeField(default=timezone.now)
-    duration = models.IntegerField(default=None, null=True)
+class TimeLog(models.Model):
+    task = models.ForeignKey(Task, on_delete=models.CASCADE, related_name='timelog_task', null=True)
+    user = models.ForeignKey(User, on_delete=models.CASCADE, related_name='timelog_user', null=True)
+    started_at = models.DateTimeField(default=timezone.now)
+    duration = models.DurationField(default=datetime.timedelta(), null=True)
+
+    class Meta:
+        ordering = ['-id']
 
     def __str__(self):
         return f'Timelog of \"{self.task.title}\" id({self.id})'
 
 
+class Timer(models.Model):
+    class Meta:
+        unique_together = [
+            ('user', 'task')
+        ]
+        ordering = ['-id']
+
+    user = models.ForeignKey(User, on_delete=models.CASCADE, related_name='user_timer')
+    task = models.ForeignKey(Task, on_delete=models.CASCADE, related_name='task_timer')
+    started_at = models.DateTimeField(auto_now_add=True)
+    is_started = models.BooleanField(default=False)
+
+    def start(self):
+        if not self.is_started:
+            self.is_started = True
+            self.started_at = timezone.now()
+            self.save()
+
+    def stop(self):
+        if self.is_started:
+            duration = timezone.now() - self.started_at
+
+            self.is_started = False
+            self.save()
+
+            timelog = TimeLog.objects.create(
+                user=self.user,
+                task=self.task,
+                started_at=self.started_at,
+                duration=duration
+            )
+            timelog.save()
+        else:
+            raise ValidationError({'detail': f'Task id:{self.id} has no ongoing timer.'})
+
+
+@receiver(post_save, sender=Task)
+def send_email(sender, instance, *args, **kwargs):
+    if instance.status == 'completed':
```

Comment:
emailul trebuie trimis doar in momentul in care statutul se schimba in completed. Cum e la tine acum o sa se transmita email la fiecare update al taskului